### PR TITLE
Create add items and expiries

### LINKED ISSUE #18: Create page for 'Add new items'
Similar to https://deck.nhhs-sjb.org/inventory/items/add

* Add a new page to 'add an item' (should be accessible from FA homepage)
* Consider using [stepper](https://mui.com/material-ui/react-stepper/) from MUI
* Some rules to follow when adding new items
  - An item either has expiry or no expiry (once an item is created as 'has expiry' it cannot be converted to 'no expiry')
  - Refer to `inventory/models`: `ItemModels` and `ItemExpiryModels` to see how DB is structured

### PR DESCRIPTION
**Pull request by cheehengk**
Pull Request Details:

Description:
Create page for adding items with and without expiries. Include api, and frontend logic and ui.
Closes #18 

Demo: [click for video](https://drive.google.com/drive/folders/1WsMaoaTOHxEaoL6EO9vRTlvBUNUwJfFe?usp=share_link)

Changes Made:
* Add add_item page (base_url/inventory/add_item)
* Add item without expiry stepper and form
* Add item with expiry stepper and form
* Add javascript to handle check for similar items already added
* Add api endpoints and serializers for handling adding items with and without expiries
* Add basic tests for most components
* Merge changes from 42aa3b560e46ee8de775f49227d497472833bfc7

Bugs reported:
#31

Regards,
Chee Heng

https://github.com/sjb-deck/deck/assets/88090694/7ee2d1cb-6f7f-4791-a1db-9691930f033f

### CODE REVIEW

File: inventory/templates/add_item.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends "inventory_base.html" %} {% load static %} {% block title %}FA
+Items{% endblock %} {% block content %}
+<script>
```
Comment (jonasgwt) [inline inventory/templates/add_item.html:3]: no need for this entire script block we will be using APIs

File: utils/submitForm.js [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+  setActiveStep,
+  setItemFormData,
+  setAddType,
+  setSuccessDialogOpen,
+  setSuccessMessage,
+  setErrorDialogOpen,
+  setLoading,
+) => {
+  setLoading(true);
+  postWithCSRF('/inventory/add_item_post', itemFormData)
```
Comment (jonasgwt) [inline utils/submitForm.js:323]: extract out the API endpoint URL into `globals.js`, i believe that the URL will change if u choose to take the suggestion in `inventory/urls.py`

File: utils/submitForm.js [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+      unit: expiryFormData.unit,
+      image: expiryFormData.image,
+      total_quantityopen: sumTotalQuantityOpen,
+      total_quantityunopened: sumTotalQuantityUnopened,
+      min_quantityopen: expiryFormData.min_quantityopen,
+      min_quantityunopened: expiryFormData.min_quantityunopened,
+    },
+    expiry: modifiedExpiry,
+  };
+  postWithCSRF('/inventory/add_expiry_post', formData)
```
Comment (jonasgwt) [inline utils/submitForm.js:403]: similar to https://github.com/sjb-deck/deck/pull/33#discussion_r1251127939

File: utils/levenshteinDistance.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
+const LENGTH_THRESHOLD = 3;
+
+/**
```
Comment (jonasgwt) [inline utils/levenshteinDistance.js:3]: dammm! insane

File: webpack.config.js [hunk 5]
```diff
@@ -13,6 +13,10 @@ module.exports = {
       './inventory/src/scss/itemIndex.scss',
     ],
     cartIndex: ['./inventory/src/js/cartIndex.js'],
+    addItem: [
+      './inventory/src/js/addItem.js',
+      './inventory/src/scss/addItem.scss',
```
Comment (jonasgwt) [inline webpack.config.js:18]: actually no need to include the scss file if it's not being used or needed, can delete the `addItem.scss` file if you feel its not going to be used in the future too

**Review by jonasgwt — COMMENTED** [review]
nice! just some comments but implementaion wise is amazing

for the images in create item form, it is best to allow users to take an image and upload the image directly in the form. But that will require some exploration to figure out how to post to Django, will put that in a seperate issue, we focus on pushing a MVP first

**Comment by cheehengk** [comment]
> nice! just some comments but implementaion wise is amazing
> 
> for the images in create item form, it is best to allow users to take an image and upload the image directly in the form. But that will require some exploration to figure out how to post to Django, will put that in a seperate issue, we focus on pushing a MVP first

Can set up GCS as default bucket storage for static files, but need bit of config and planning

**Review by jonasgwt — APPROVED** [review]

**Pull request merged**